internal-services/catalog/pulp-push-disk-images-task.yaml

```diff
@@ -0,0 +1,191 @@
+          valueFrom:
+            secretKeyRef:
+              name: $(params.udcacheSecret)
+              key: TODO
+        - name: udcacheKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.udcacheSecret)
+              key: TODO
+        - name: "SNAPSHOT_JSON"
+          value: "$(params.snapshot_json)"
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+
+        exitfunc() {
+            local err=$1
+            local line=$2
+            local command="$3"
+            if [ $err -eq 0 ] ; then
+                echo -n "Success" > $(results.result.path)
+            else
+                echo -n "$0: ERROR '$command' failed at line $line - exited with status $err" > $(results.result.path)
+            fi
+            exit 0 # exit the script cleanly as there is no point in proceeding past an error or exit call
+        }
+        # due to set -e, this catches all EXIT and ERR calls and the task should never fail with nonzero exit code
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        # Setup required variables
+        export EXODUS_GW_CERT=/tmp/exodus.crt
+        export EXODUS_GW_KEY=/tmp/exodus.key
+        export PULP_CERT=/tmp/pulp.crt
+        export PULP_KEY=/tmp/pulp.key
+        export UDCACHE_CERT=/tmp/udc.crt
+        export UDCACHE_KEY=/tmp/udc.key
+        export EXODUS_GW_ENV=$(params.exodusGwEnv)
+        export EXODUS_GW_URL="$exodusGwUrl"
+        export EXODUS_PULP_HOOK_ENABLED=True
+        export EXODUS_GW_TIMEOUT=7200
+
+        set +x
+        echo "$exodusGwCert" > "$EXODUS_GW_CERT"
+        echo "$exodusGwKey" > "$EXODUS_GW_KEY"
+        echo "$pulpCert" > "$PULP_CERT"
+        echo "$pulpKey" > "$PULP_KEY"
+        echo "$udcacheCert" > "$UDCACHE_CERT"
+        echo "$udcacheKey" > "$UDCACHE_KEY"
+        set -x
+
+        DISK_IMAGE_DIR="$(mktemp -d)"
+
+        process_component() { # Expected argument is [component json]
+            COMPONENT=$1
+            PULLSPEC=$(jq -r '.containerImage' <<< "${COMPONENT}")
+            DESTINATION="${DISK_IMAGE_DIR}/$(jq -r '.staged.destination' <<< "${COMPONENT}")/FILES"
+            mkdir -p "${DESTINATION}"
+            DOWNLOAD_DIR=$(mktemp -d)
+            cd $DOWNLOAD_DIR
+            # oras has very limited support for selecting the right auth entry,
+            # so create a custom auth file with just one entry
+            AUTH_FILE=$(mktemp)
+            select-oci-auth "${PULLSPEC}" > "$AUTH_FILE"
+            oras pull --registry-config "$AUTH_FILE" "$PULLSPEC"
+            NUM_MAPPED_FILES=$(jq '.staged.files | length' <<< ${COMPONENT})
+            for ((i = 0; i < $NUM_MAPPED_FILES; i++)) ; do
+                FILE=$(jq -c --arg i "$i" '.staged.files[$i|tonumber]' <<< $COMPONENT)
+                SOURCE=$(jq -r '.source' <<< $FILE)
+                FILENAME=$(jq -r '.filename' <<< $FILE)
+                gzip -d "${SOURCE}.gz" || echo "didn't find mapped file: ${SOURCE}.gz"
```

Comment:
Shouldn't this result in an error? I am not sure right how the `staged.files` array is generated. But surely all the files should be in the oci artifact, no?

Reply:
https://github.com/konflux-ci/release-service-catalog/pull/463#discussion_r1661627128
Pretty sure I talked to David about it on slack

Reply:
Fair enough.